frontend/src/components/ChatMessage.vue

```diff
@@ -1,40 +1,58 @@
 <script setup lang="ts">
-import { messageService } from '@/services/message.service';
+// External libraries
 import { ref } from 'vue';
-const { messageSuccess, messageError } = messageService();
+import { useI18n } from 'vue-i18n';
 
-const props = defineProps<{
-  message: string;
-  isSent: boolean;
-}>();
+// Internal dependencies
+import { messageService } from '@/services/message.service';
+import { type MessageWrapper } from '../types/wrapper';
 
-const { message, isSent } = props;
+/**
+ * Props for ChatMessage component.
+ */
+const props = defineProps<MessageWrapper>();
 
+const { t } = useI18n();
+const { message, isSent } = props;
+const { messageSuccess, messageError } = messageService();
 const isCopying = ref(false);
 
-const copyToClipboard = (event) => {
-  // Interrompo se c'è già una copia in corso
-  if (isCopying.value) return;
-  isCopying.value = true;
-  // Risalgo al messaggio più vicino al bottone cliccato
-  const messageContent = event.currentTarget.closest('.message').querySelector('p');
-  if (!messageContent) return;
-  // Estraggo il contenuto dall'elemento ed elimino eventuali spazi all'inizio e alla fine della stringa
-  const text = messageContent.textContent.trim();
+/**
+ * Copy the provided text to the clipboard.
+ * @param text - The text to copy to the clipboard.
+ */
+const performCopy = (text: string) => {
   navigator.clipboard
     .writeText(text)
     .then(() => {
-      console.log('Text copied to clipboard: ', text);
-      messageSuccess('Copy', 'Text copied to clipboard');
+      console.log(t('general.clipboard.success') + ': ', text);
+      messageSuccess(t('general.clipboard.name'), t('general.clipboard.success'));
       setTimeout(() => {
         isCopying.value = false;
       }, 2000);
     })
     .catch((err) => {
-      console.error('Error when copying to clipboard: ', err);
-      messageError('Copy', 'Error when copying to clipboard');
+      console.error(t('general.clipboard.error') + ': ', err);
+      messageError(t('general.clipboard.name'), t('general.clipboard.error'));
```

Comment:
```suggestion
      messageError(t('general.clipboard.name'), t('general.clipboard.error'));
```
toglierei i messaggi in console essendo ridondanti dall'interfaccia

Reply:
ok